doc/common/Cadl-Project-Scripts.md

```diff
@@ -0,0 +1,127 @@
+# Integrating with cadl sync and generate scripts
+
+There are 2 common scripts provided for each language to be able to generate from within the language
+repo and use the remote cadl definition in the spec repo.
+
+## Prerequisites
+
+There are 4 things that these two scripts expect are set up in your language repo before they will run correctly.
+
+### cadl-location.yaml
+
+This file should live under the project directory for each service and has the following properties
+
+| Property | Description | IsRequired |
+| --- | --- | --- |
+| directory | The top level directory where the main.cadl for the service lives.  This should be relative to the spec repo root such as `specification/cognitiveservices/OpenAI.Inference` | true |
+| additionalDirectories | Sometimes a cadl file will use a relative import that might not be under the main directory.  In this case you can specify additional directories as a list to sync so that all needed files are synced. | false: default = null |
```

Comment:
Don't know if this is a feedback, but I had to read this several times to finally understand this is related to your sparse checkout, that you need to configure git to explain the external folders to get. I was quite confused, as technically from a pure git clone this wouldn't be necessary. Again, I don't know if this is just me, or if we could rephrase to clarify.